### PR TITLE
fix(sortable-table): refactor sortable-table to rerender with data change

### PR DESCRIPTION
**Pull request by ctaymor**
New interface for sortable-table, which also fixes the bug where the
table didn't rerender when the data changed.

[Finishes #97658788]

BREAKING CHANGE: (javascript) React component has new wrapping components for
the header and rows.

### CODE REVIEW

**Comment by stubbornella** [comment]
@gpleiss - I'd like for both of use to code review this one, do you mind?

File: src/pivotal-ui/javascripts/components.js [hunk 1]
```diff
@@ -120,5 +118,4 @@ module.exports = {
   BackToTop: require('pui-react-back-to-top').BackToTop,
 
   PortalSource: require('pui-react-portals').PortalSource,
-  PortalDestination: require('pui-react-portals').PortalDestination
-};
+  PortalDestination: require('pui-react-portals').PortalDestination, ...sortableTable};
```
Comment (gpleiss) [inline src/pivotal-ui/javascripts/components.js]: eww... why is this all on one line? :)

File: src/pivotal-ui-react/sortable-table/sortable-table.js [hunk 2]
```diff
@@ -44,100 +60,91 @@ var TableRow = React.createClass({
...
     });
+  },
+
+  render() {
+    let {className, style, id} = this.props;
+    let rows = this.sortedRows();
 
     return (
-      <table className={classnames('table', 'table-sortable', classes)}>
+      <table className={classnames('table', 'table-sortable', className)} id={id} style={style} >
```
Comment (gpleiss) [inline src/pivotal-ui-react/sortable-table/sortable-table.js]: Could be refactored to use our mergeProps helper, but not important.

File: src/pivotal-ui-react/sortable-table/sortable-table.js [hunk 3]
```diff
@@ -44,100 +60,91 @@ var TableRow = React.createClass({
...
-    var {columns, data} = this.props;
-    var column = columns[0].name;
-    return {
-      sort: {
-        column,
-        order: 'asc'
-      },
-      data: sortBy(data, column)
-    };
+    return {sortColumn: 0, sortAscending: true};
```
Comment (gpleiss) [inline src/pivotal-ui-react/sortable-table/sortable-table.js]: What if the first column is not sortable?

Comment (gpleiss) [inline src/pivotal-ui-react/sortable-table/sortable-table.js]: Also, a user should be able to choose the default sort column/order. Not sure exactly how this should look.

**Comment by gpleiss** [comment]
So before this PR can be merged we need to address the non-sortable first column bug. Then this should be ready for merge IMO.

**Pull request merged**